File: WhichPharmaPortal/WhichPharma.Models/Client/RFQRequest.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using WhichPharma.Models.Attributes;

namespace WhichPharmaPortal.Models.Client
{
    public class RFQRequest
    {
        [Optional]
        [StringLength(500, MinimumLength = 4)]
        public string SenderEmail { get; set; }
        [Optional]
        [MaxLength(500)]
        public string SenderPassword { get; set; }
        [Optional]
        [MaxLength(500)]
        public string SenderGmailAccessCode { get; set; }
        [Required]
        [MinLength(1)]
        public IEnumerable<RFQEmailData> EmailsData { get; set; }
        [Optional]
        public IEnumerable<string> RfqNumbersToAssignToExistingOnes { get; set; }
    }
}

File: WhichPharmaPortal/WhichPharmaPortal/ClientApp/src/models/RFQRequest.ts
import { RFQEmailData } from './RFQEmailData';
import { RFQState } from './RFQState';

export interface RFQRequest {
    senderEmail?: string;
    senderPassword?: string;
    senderGmailAccessCode?: string;
    emailsData: RFQEmailData[];
    rfqNumbersToAssignToExistingOnes?: string[];
}

export interface IRFQDueRminderRequest {
    reminder: number;
    dueDate: string;
}

export interface IRFQStateReasonRequest {
    reason: string;
    state: RFQState;
}
